v2/api/dbformysql/v1api20210501/storage/flexible_server_types_overrides.go

```diff
@@ -32,7 +33,11 @@ func (server *FlexibleServer_Spec) AssignPropertiesTo(dst *v20230630s.FlexibleSe
 	// Get the current property bag
 	pb := genruntime.NewPropertyBag(dst.PropertyBag)
 
-	if server.SourceServerResourceId != nil {
+	// If we only have the legacy property, populate the new one from it
```

Comment:
Is this right? Does this RoundTrip correctly?

If old has `ResourceId` set, so it flows to new `ServerResourceReference` and back to `ResourceId` (based on the `AssignPropertiesFrom` above), which si good.
but if old gets `ResourceReference` set, it will flow to `ResourceReference` as well, and then back to `ResourceId` which isn't correct, isn't that right?

Reply:
See my comment on azure-arm.yaml about if we even want to change this old API version at all.